File: JlgCommon/Extensions/DoubleExtensions.cs
using System;

namespace JlgCommon.Extensions
{
    public static class DoubleExtensions
    {
        public static double? DifferenceInPercents(this double nr1, double nr2)
        {
            if (Math.Abs(nr2) > 0)
            {
                return Math.Round((nr1 * 100 / nr2) - 100, 2);
            }
            else
            {
                return null;
            }
        }

        public static double? DifferenceInPercents(this double? nr1, double? nr2)
        {
            if (!nr1.HasValue || !nr2.HasValue)
            {
                return null;
            }

            if (Math.Abs(nr2.Value) > 0)
            {
                return Math.Round((nr1.Value * 100 / nr2.Value) - 100, 2);
            }
            else
            {
                return null;
            }
        }

        public static double IncreaseByPercent(this double nr, double percent)
        {           
            return nr + nr.PercentValue(percent);
        }

        public static double PercentValue(this double nr, double percent)
        {
            var percentValueFromNr = (percent / 100) * nr;
            return percentValueFromNr;
        }      

    }
}

File: Java/JlgCommon/src/extensions/DoubleExtensions.java
package extensions;
import java.text.DecimalFormat;
/**
 * Created by Dan on 07/09/2015.
 */
public final class DoubleExtensions {

    public static double roundToDecimals(double nr, int decimals){

        StringBuilder decimalFormatSb = new StringBuilder();
        decimalFormatSb.append("0.");
        for (int i=0; i<decimals; i++) {
            decimalFormatSb.append("0");
        }
        DecimalFormat df = new DecimalFormat(decimalFormatSb.toString());
        String formatedDouble = df.format(nr);
        return Double.valueOf(formatedDouble);
    }

    public static double differenceInPercents(double nr1, double nr2)
    {
        if (Math.abs(nr2) > 0) {
            return roundToDecimals(nr1 * 100 / nr2 - 100, 2);
        }
        else {
            return Double.NaN;
        }
    }

    public static double percentValue(double nr, double percent)
    {
        double percentValueFromNr = (percent / 100) * nr;
        return percentValueFromNr;
    }

    public static double increaseByPercent(double nr, double percent)
    {
        return nr + percentValue(nr, percent);
    }

    private DoubleExtensions(){
    }
}
